packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+// @ts-check
+
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { fit, M, makeCopyBag, makeStore } from '@agoric/store';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X } = assert;
+
+export const KW = harden({
+  /** seat keyword for use in offers to return an attestation. */
+  Attestation: 'Attestation',
+});
+
+/**
+ * Find-or-create value in store.
+ *
+ * @type {<K, V>(store: Store<K, V>, key: K, thunk: () => V) => V}
+ */
+const getOrElse = (store, key, make) => {
+  if (store.has(key)) {
+    return store.get(key);
+  }
+  const value = make();
+  store.init(key, value);
+  return value;
+};
+
+/**
+ * To support `makeAttestation`, we directly mint attestation payments.
+ * We still use a ZCFMint because returning attestations is done
+ * through Zoe offers.
+ *
+ * @param {ZCF} zcf
+ * @param {ZCFMint} zcfMint
+ * @param {Amount} amountToMint
+ * @returns {Promise<Payment>}
+ */
+const mintZCFMintPayment = (zcf, zcfMint, amountToMint) => {
+  const { userSeat, zcfSeat } = zcf.makeEmptySeatKit();
+  zcfMint.mintGains(harden({ [KW.Attestation]: amountToMint }), zcfSeat);
+  zcfSeat.exit();
+  return E(userSeat).getPayout(KW.Attestation);
+};
+
+/**
+ * Make an ZCFMint attentuated so that minting represents
+ * putting a lien on staked assets. Payments can be returned,
+ * releasing (part of) the lien.
+ *
+ * @param {ZCF} zcf
+ * @param {string} keyword for use in makeZCFMint
+ * @param {Brand<'nat'>} stakeBrand brand of the staked assets
+ * @param {ERef<StakingAuthority>} lienBridge bridge to account state
+ */
+const makeAttestationKit = async (zcf, keyword, stakeBrand, lienBridge) => {
+  const { add, subtract, makeEmpty, isGTE, coerce } = AmountMath;
+  const empty = makeEmpty(stakeBrand);
+  const zcfMint = await zcf.makeZCFMint(keyword, AssetKind.COPY_BAG);
+  const { issuer, brand: attBrand } = zcfMint.getIssuerRecord();
+
+  /**
+   * @param {string} address
```

Comment:
what kind of address? is there any validation to do of the value?

Reply:
This design relies on `lienBridge` to validate addresses. As far as this layer is concerned, they're just strings.

Reply:
As far as the code reader is concerned, what kind of address is it? :)  I get that its runtime type is `string` and that it represents an "address". Is it a Cosmos account address? I don't think that all needs to go in the symbol names but consider,
```suggestion
   * @param {string} address Cosmos account
```

In the future we may want to make this an opaque type because it's not meant to do all the things a string can do (e.g. `concat`).

Reply:
It's a Cosmos account iff the `lienBridge` is a bridge to Cosmos.

Actually, we do have an `Address` type. I'll use that and clarify it a bit.